public/js/psychologistForm.js

```diff
@@ -0,0 +1,296 @@
+      });
+  }
+
+  form.onsubmit = (event) => {
+    event.preventDefault();
+    saveButton.disabled = true;
+
+    let url;
+    const options = {
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify({
+        firstName: firstNameInput.value,
+        lastName: lastNameInput.value,
+        username: usernameInput.value,
+        password: passwordInput.value,
+        email: emailInput.value,
+        phone: parseInt(phoneInput.value, 10),
+        address: addressInput.value,
+        availability: {
+          monday: {
+            availability: !!mondayAvailability.value,
+            from: parseInt(mondayFromInput.value, 10),
+            to: parseInt(mondayToInput.value, 10),
+          },
+          tuesday: {
+            availability: !!tuesdayAvailability.value,
+            from: parseInt(tuesdayFromInput.value, 10),
+            to: parseInt(tuesdayToInput.value, 10),
+          },
+          wednesday: {
+            availability: !!wednesdayAvailability.value,
+            from: parseInt(wednesdayFromInput.value, 10),
+            to: parseInt(wednesdayToInput.value, 10),
+          },
+          thursday: {
+            availability: !!thursdayAvailability.value,
+            from: parseInt(thursdayFromInput.value, 10),
+            to: parseInt(thursdayToInput.value, 10),
+          },
+          friday: {
+            availability: !!fridayAvailability.value,
+            from: parseInt(fridayFromInput.value, 10),
+            to: parseInt(fridayToInput.value, 10),
+          },
+          saturday: {
+            availability: !!saturdayAvailability.value,
+            from: parseInt(saturdayFromInput.value, 10),
+            to: parseInt(saturdayToInput.value, 10),
+          },
+          sunday: {
+            availability: !!sundayAvailability.value,
+            from: parseInt(sundayFromInput.value, 10),
+            to: parseInt(sundayToInput.value, 10),
+          },
+        },
+      }),
+    };
+
+    if (paramsId) {
+      options.method = 'PUT';
+      url = `${window.location.origin}/api/psychologists/${paramsId}`;
+    } else {
+      options.method = 'POST';
+      url = `${window.location.origin}/api/psychologists`;
+    }
+
+    fetch(url, options)
+      .then((response) => {
+        if (response.status !== 200 && response.status !== 201) {
+          return response.json().then(({ message }) => {
+            throw new Error(message);
+          });
+        }
+        return response.json();
+      })
+      .then(() => {
+        window.location.href = `${window.location.origin}/views/psychologistList.html`;
+      })
+      .catch((error) => {
+        errorMessage.innerText = error;
+      })
+      .finally(() => {
+        saveButton.disabled = false;
+      });
+  };
+
+  mondayAvailability.onblur = () => {
+    if (mondayAvailability.value === 'false') {
+      mondayFromInput.value = 0;
+      mondayToInput.value = 0;
```

Comment:
I do not think it is necessary to set the value to 0, since the day is false, it is assumed that the hours are not taken into account

Reply:
The problem is that those input are required so if i left the input empty the validation will not pass.